AttributeError: 'Flask' object has no attribute 'login_manager' FAILED

```py
=========================== short test summary info ============================
FAILED api/tests/unit_tests/controllers/console/auth/test_data_source_bearer_auth.py::TestApiKeyAuthDataSource::test_get_api_key_auth_data_source - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/auth/test_data_source_bearer_auth.py::TestApiKeyAuthDataSource::test_get_api_key_auth_data_source_empty - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/auth/test_data_source_bearer_auth.py::TestApiKeyAuthDataSourceBinding::test_create_binding_successful - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/auth/test_data_source_bearer_auth.py::TestApiKeyAuthDataSourceBinding::test_create_binding_failure - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/auth/test_data_source_bearer_auth.py::TestApiKeyAuthDataSourceBindingDelete::test_delete_binding_successful - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/auth/test_data_source_oauth.py::TestOAuthDataSourceSync::test_sync_successful - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/auth/test_oauth_server.py::TestOAuthServerUserAuthorizeApi::test_successful_authorize - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/core/repositories/test_sqlalchemy_workflow_node_execution_repository.py::test_to_db_model_requires_constructor_context - AttributeError: type object 'str' has no attribute 'LLM'
FAILED api/tests/unit_tests/core/repositories/test_sqlalchemy_workflow_node_execution_repository.py::test_to_db_model_requires_creator_user_id_and_role - AttributeError: type object 'str' has no attribute 'LLM'
FAILED api/tests/unit_tests/core/repositories/test_sqlalchemy_workflow_node_execution_repository.py::test_is_duplicate_key_error_and_regenerate_id - AttributeError: type object 'str' has no attribute 'LLM'
FAILED api/tests/unit_tests/core/repositories/test_sqlalchemy_workflow_node_execution_repository.py::test_to_domain_model_loads_offloaded_files - AttributeError: type object 'str' has no attribute 'LLM'
FAILED api/tests/unit_tests/core/repositories/test_sqlalchemy_workflow_node_execution_repository.py::test_to_domain_model_returns_early_when_no_offload_data - AttributeError: type object 'str' has no attribute 'LLM'
FAILED api/tests/unit_tests/core/repositories/test_sqlalchemy_workflow_node_execution_repository.py::test_save_execution_data_handles_existing_db_model_and_truncation - AttributeError: type object 'str' has no attribute 'LLM'
FAILED api/tests/unit_tests/core/repositories/test_sqlalchemy_workflow_node_execution_repository.py::test_save_execution_data_truncates_outputs_and_process_data - AttributeError: type object 'str' has no attribute 'LLM'
FAILED api/tests/unit_tests/core/repositories/test_sqlalchemy_workflow_node_execution_repository.py::test_save_execution_data_handles_missing_db_model - AttributeError: type object 'str' has no attribute 'LLM'
FAILED api/tests/unit_tests/core/repositories/test_sqlalchemy_workflow_node_execution_repository.py::test_save_retries_duplicate_and_logs_non_duplicate - AttributeError: type object 'str' has no attribute 'LLM'
FAILED api/tests/unit_tests/core/repositories/test_sqlalchemy_workflow_node_execution_repository.py::test_save_logs_and_reraises_on_unexpected_error - AttributeError: type object 'str' has no attribute 'LLM'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_chat_message_list_not_found - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_chat_message_list_success - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_feedback_not_found - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_feedback_success - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_annotation_count - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_suggested_questions_success - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_suggested_questions_errors[MessageNotExistsError-NotFound] - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_suggested_questions_errors[ConversationNotExistsError-NotFound] - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_suggested_questions_errors[ProviderTokenNotInitError-ProviderNotInitializeError] - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_suggested_questions_errors[QuotaExceededError-ProviderQuotaExceededError] - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_suggested_questions_errors[ModelCurrentlyNotSupportError-ProviderModelCurrentlyNotSupportError] - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_suggested_questions_errors[SuggestedQuestionsAfterAnswerDisabledError-AppSuggestedQuestionsAfterAnswerDisabledError] - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_suggested_questions_errors[Exception-InternalServerError] - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_feedback_export_success - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_message.py::TestMessageEndpoints::test_message_api_get_success - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_statistic.py::TestStatisticEndpoints::test_daily_message_statistic - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_statistic.py::TestStatisticEndpoints::test_daily_conversation_statistic - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_statistic.py::TestStatisticEndpoints::test_daily_terminals_statistic - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_statistic.py::TestStatisticEndpoints::test_daily_token_cost_statistic - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_statistic.py::TestStatisticEndpoints::test_average_session_interaction_statistic - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_statistic.py::TestStatisticEndpoints::test_user_satisfaction_rate_statistic - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_statistic.py::TestStatisticEndpoints::test_average_response_time_statistic - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_statistic.py::TestStatisticEndpoints::test_tokens_per_second_statistic - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_statistic.py::TestStatisticEndpoints::test_invalid_time_range - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_statistic.py::TestStatisticEndpoints::test_time_range_params_passed - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_workflow_variable_collection_get_success - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_workflow_variable_collection_get_not_exist - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_workflow_variable_collection_delete - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_node_variable_collection_get_success - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_node_variable_collection_get_invalid_node_id - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_node_variable_collection_delete - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_variable_api_get_success - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_variable_api_get_not_found - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_variable_api_patch_success - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_variable_api_delete_success - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_variable_reset_api_put_success - AttributeError: 'Flask' object has no attribute 'login_manager'
specified configuration file /home/runner/work/dify/dify/api/tests/integration_tests/vdb.env not exist
2026-03-23 03:36:23,840.840 INFO [MainThread] [conftest.py:73]  - DifyTestContainers initialized - ready to manage test containers
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_conversation_variable_collection_get - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_system_variable_collection_get - AttributeError: 'Flask' object has no attribute 'login_manager'
FAILED api/tests/unit_tests/controllers/console/app/test_workflow_draft_variable.py::TestWorkflowDraftVariableEndpoints::test_environment_variable_collection_get - AttributeError: 'Flask' object has no attribute 'login_manager'
```

@dosu how to fix it.

--- a/api/pyproject.toml
+++ b/api/pyproject.toml
@@ -79,6 +79,7 @@ dependencies = [
     "tiktoken~=0.12.0",
     "transformers~=5.3.0",
     "unstructured[docx,epub,md,ppt,pptx]~=0.21.5",
+    "pypandoc~=1.13",
     "yarl~=1.23.0",
     "webvtt-py~=0.5.1",
     "sseclient-py~=1.9.0",

--- a/api/tests/unit_tests/core/repositories/test_sqlalchemy_workflow_node_execution_repository.py
+++ b/api/tests/unit_tests/core/repositories/test_sqlalchemy_workflow_node_execution_repository.py
@@ -24,7 +24,7 @@ from core.repositories.sqlalchemy_workflow_node_execution_repository import (
 )
 from dify_graph.entities import WorkflowNodeExecution
 from dify_graph.enums import (
-    NodeType,
+    BuiltinNodeTypes,
     WorkflowNodeExecutionMetadataKey,
     WorkflowNodeExecutionStatus,
 )
@@ -67,7 +67,7 @@ def _execution(
         index=1,
         predecessor_node_id=None,
         node_id="node-id",
-        node_type=NodeType.LLM,
+        node_type=BuiltinNodeTypes.LLM,
         title="Title",
         inputs=inputs,
         outputs=outputs,
@@ -387,7 +387,7 @@ def test_to_domain_model_loads_offloaded_files(monkeypatch: pytest.MonkeyPatch) 
     db_model.index = 1
     db_model.predecessor_node_id = None
     db_model.node_id = "node"
-    db_model.node_type = NodeType.LLM
+    db_model.node_type = BuiltinNodeTypes.LLM
     db_model.title = "t"
     db_model.inputs = json.dumps({"trunc": "i"})
     db_model.process_data = json.dumps({"trunc": "p"})
@@ -441,7 +441,7 @@ def test_to_domain_model_returns_early_when_no_offload_data(monkeypatch: pytest.
     db_model.index = 1
     db_model.predecessor_node_id = None
     db_model.node_id = "node"
-    db_model.node_type = NodeType.LLM
+    db_model.node_type = BuiltinNodeTypes.LLM
     db_model.title = "t"
     db_model.inputs = json.dumps({"i": 1})
     db_model.process_data = json.dumps({"p": 2})

